packages/types/index.ts

```diff
@@ -0,0 +1,161 @@
+import type { AsyncIterator } from 'asynciterator';
+import type { Map } from 'immutable';
+import type * as RDF from 'rdf-js';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
+ * An immutable solution mapping object.
+ * This maps variables to a terms.
+ *
+ * Variables are represented as strings containing the variable name prefixed with '?'.
+ * Blank nodes are represented as strings containing the blank node name prefixed with '_:'.
+ * Terms are named nodes, literals or the default graph.
+ */
+export type Bindings = Map<string, RDF.Term>;
+
+/**
+ * A stream of bindings.
+ *
+ * Next to the list of available variables,
+ * an optional metadata hash can be present.
+ *
+ * @see Bindings
+ */
+export type BindingsStream = AsyncIterator<Bindings>;
+
+/**
+ * Query operation output for a bindings stream.
+ * For example: SPARQL SELECT results
+ */
+export interface IActorQueryOperationOutputBindings extends IActorQueryOperationOutputStream {
+  /**
+   * The type of output.
+   */
+  type: 'bindings';
+  /**
+   * The stream of bindings resulting from the given operation.
+   */
+  bindingsStream: BindingsStream;
+  /**
+   * The list of variable names (without '?') for which bindings are provided in the stream.
+   */
+  variables: string[];
+  /**
+   * If any of the bindings could contain an undefined variable binding.
+   * If this is false, then all variables are guaranteed to have a defined bound value in the bindingsStream.
+   */
+  canContainUndefs: boolean;
+}
+
+/**
+ * Query operation output for quads.
+ * For example: SPARQL CONSTRUCT results
+ */
+export interface IActorQueryOperationOutputQuads extends IActorQueryOperationOutputStream {
+  /**
+   * The type of output.
+   */
+  type: 'quads';
+  /**
+   * The stream of quads.
+   */
+  quadStream: RDF.Stream & AsyncIterator<RDF.Quad>;
+}
+
+/**
+ * Query operation output for quads.
+ * For example: SPARQL ASK results
+ */
+export interface IActorQueryOperationOutputBoolean extends IActorQueryOperationOutputBase {
+  /**
+   * The type of output.
+   */
+  type: 'boolean';
+  /**
+   * A promise resolving to the boolean output of the operation.
+   */
+  booleanResult: Promise<boolean>;
+
+}
+
+/**
+ * Binds a quad pattern term's position to a variable.
+ */
+export type PatternBindings = Record<string, RDF.Variable>;
+
+export interface IActionQueryOperation extends IAction {
+  /**
+   * The query operation to handle.
+   */
+  operation: Algebra.Operation;
+}
+
+/**
+ * Query operation output.
+ * @see IActorQueryOperationOutputBindings, IActorQueryOperationOutputQuads, IActorQueryOperationOutputBoolean
+ */
+export type ActorQueryOperationOutput =
```

Comment:
```suggestion
export type IActorQueryOperationOutput =
```

I'm aware this is not an interface, but I would do this to be better in line with the currently existing interface/type (so that developers only have to change the import target, not the rest of their code).

We may want to add a `TODO` comment above this to properly rename upon our next major update. Perhaps we could then even use a name such as `QueryOutput`.

Reply:
Done!